application/libraries/ObjectPatch/Patcher.php

```diff
@@ -80,5 +84,65 @@ private function handleOp(OpInterface $op): void
                 )
             );
         }
+        return $returnedData;
+    }
+
+    /**
+     * Moves mappings of tempIds and their real ids of same object types
+     * together for a more structured return to the client.
+     * @param array $returnedData
+     * @param int $operationsApplied
+     * @return array
+     */
+    private function reorganizeReturnedData(
```

Comment:
This method refers to concepts which are specific to our use of the patcher in PatcherSurvey (question, subquestion, answer) but this Patcher class is meant to be generic. Either this functionality belongs in PatcherSurvey or it should be modified so that the specific use case are passed in as an argument. For example you can pass in `$knownMaps` as a property.